feature(guidance-scale-input-field): adds input field for guidance scale

stbaione opened the pull request:
# Description

This PR creates a new `TextToImageModelSettings` component, and adds an input field for `GuidanceScale`.

[guidance_scale](https://medium.com/@wangdk93/stablediffusion-guidance-scale-1822cfff7c9d) is a text-to-image parameter that controls how closely the generated image adheres to the text prompt.

From @monorimet:

```text
Precision: 2 decimals is the norm
Min/Max: Depends on use case, but generally 0-20
```
> [!NOTE]
> Users will typically only use a small subset of those values.

For the default, we previously had it set to `7.5`, so I just kept that value.

Since we were adding another setting, seemed like a good opportunity to modularize.
Moved the `DiscreteSlider` to `TextToImageModelSettings` and setup bindings between component and page.

# Implementation References

## Normal full-screen
![image](https://github.com/user-attachments/assets/ad668fd3-a015-4dc3-9192-f978de4e9a45)

## Functional Diff
![image](https://github.com/user-attachments/assets/a3605fd1-077d-44b1-9b4f-1d718a748026)

> [!NOTE]
> This is basically saying, "Don't pay attention to the prompt at all"
![image](https://github.com/user-attachments/assets/535a5c45-62c2-48da-b57b-c5a2e53be98a)


## Narrow
![image](https://github.com/user-attachments/assets/f22be48c-8001-4c70-a5f1-98e55042d2a5)

## Squeezed
![image](https://github.com/user-attachments/assets/fb07b745-e7b8-48d9-ac1f-96ad7fd9ef72)

## Mobile
Appends to scrolling content in mobile view
![image](https://github.com/user-attachments/assets/3aa5b0b0-c71d-4098-ac06-74e8cd465353)

dependabot[bot] committed:
chore(deps-dev): bump eslint-plugin-cypress from 5.0.0 to 5.0.1 (#291)

Bumps [eslint-plugin-cypress](https://github.com/cypress-io/eslint-plugin-cypress) from 5.0.0 to 5.0.1.
- [Release notes](https://github.com/cypress-io/eslint-plugin-cypress/releases)
- [Commits](https://github.com/cypress-io/eslint-plugin-cypress/compare/v5.0.0...v5.0.1)

---
updated-dependencies:
- dependency-name: eslint-plugin-cypress
  dependency-version: 5.0.1
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump typescript-eslint from 8.33.0 to 8.33.1 (#289)

Bumps [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/typescript-eslint) from 8.33.0 to 8.33.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/typescript-eslint/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.33.1/packages/typescript-eslint)

---
updated-dependencies:
- dependency-name: typescript-eslint
  dependency-version: 8.33.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump zod from 3.25.48 to 3.25.49 (#290)

Bumps [zod](https://github.com/colinhacks/zod) from 3.25.48 to 3.25.49.
- [Release notes](https://github.com/colinhacks/zod/releases)
- [Commits](https://github.com/colinhacks/zod/compare/v3.25.48...v3.25.49)

---
updated-dependencies:
- dependency-name: zod
  dependency-version: 3.25.49
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

bjacobgordon committed:
fix(text-to-image): ensures "failed to fetch" error are properly identified (#292)

bjacobgordon committed:
types(text-to-image): avoids resolving `TextToImage.Output['image']` to an object literal (#293)

- allows intellisense to display type name rather just its structure upon hover

bjacobgordon committed:
refactor(text-to-image): preps page to handle error state (#294)

- extracts subcomponent
- orders state-based components according to typical UI states: empty -> loading -> error -> partial -> complete
  - the page already handles "empty", "loading", and "complete"
  - "partial" doesn't apply there's only one element: the image
  - "error" is the hole we're trying to plug

bjacobgordon committed:
fix(library/vue): ensures that results are indeterminate for ongoing process (#295)

Clarifies that `null` is contextual according to the UI states:
- "empty": `result: null` because the process has never been initiated since last reset
- "loading": `result: null` because it isn't done yet, regardless of whether we have something in `result`
- "complete": `result: SomeProduct` when successful

bjacobgordon committed:
refactor(library/Attempt): makes it easier to build callbacks for factories (#296)

* avoids unnecessary picking of factory exports
* exposes callback types for `Attempt.that` and `Attempt.thatEventually`

stbaione committed:
feature(guidance-scale-input-field): creates a new `TextToImageModelSettings` component,
and adds an input field for `GuidanceScale`.

guidance_scale is a text-to-image parameter that controls how closely the generated image adheres to the text prompt.

Since we were adding another setting, seemed like a good opportunity to modularize.
Moved the `DiscreteSlider` to `TextToImageModelSettings` and setup bindings between component and page.

stbaione committed:
feature(guidance-scale-input-field) `sliderProps` => `diffusionStepProps`

monorimet commented:
Open to other opinions here but "guidance scale for generation" seems too verbose. If we go by what's used, comfyui uses "cfg", auto1111 uses "cfg scale" -- I think we can just use "cfg scale" as well, but we don't need to conform and could say "prompt strength (cfg scale)" or something foolproof as such.

monorimet reviewed (approved):
Overall looks good to me -- @bjacobgordon for code owner review.
One comment about a subjective thing to consider.

stbaione commented:
> Open to other opinions here but "guidance scale for generation" seems too verbose. If we go by what's used, comfyui uses "cfg", auto1111 uses "cfg scale" -- I think we can just use "cfg scale" as well, but we don't need to conform and could say "prompt strength (cfg scale)" or something foolproof as such.

Let's go with `cfg scale`. Will likely be what users are most accustomed too

stbaione committed:
Change label to `Cfg Scale`

bjacobgordon committed:
prefeature(library/vue): makes it more obvious that error UI is missing (#298)

- replaces `useStatefulProcess` composable with error-safe version
- the benefit: template-level "force unwrap" that just _begs_ for an error UI

bjacobgordon committed:
feature(text-to-image): reports offending origin in `Server.ConnectionError` (#299)

- surfaces failing origin to global error handler

bjacobgordon committed:
feature(text-to-image): adds custom alert for `Server.ConnectionError` (#300)

- before, instances of this error would propagate all the way to global handler, instructing users to file an issue
- since this error is one that the user can address themselves with a little guidance, we added an alert to present the relevant troubleshooting instructions

dependabot[bot] committed:
chore(deps-dev): bump vitest from 3.2.0 to 3.2.1 (#308)

Bumps [vitest](https://github.com/vitest-dev/vitest/tree/HEAD/packages/vitest) from 3.2.0 to 3.2.1.
- [Release notes](https://github.com/vitest-dev/vitest/releases)
- [Commits](https://github.com/vitest-dev/vitest/commits/v3.2.1/packages/vitest)

---
updated-dependencies:
- dependency-name: vitest
  dependency-version: 3.2.1
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump cypress from 14.4.0 to 14.4.1 (#305)

Bumps [cypress](https://github.com/cypress-io/cypress) from 14.4.0 to 14.4.1.
- [Release notes](https://github.com/cypress-io/cypress/releases)
- [Changelog](https://github.com/cypress-io/cypress/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/cypress-io/cypress/compare/v14.4.0...v14.4.1)

---
updated-dependencies:
- dependency-name: cypress
  dependency-version: 14.4.1
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump eslint-plugin-cypress from 5.0.1 to 5.1.0 (#306)

Bumps [eslint-plugin-cypress](https://github.com/cypress-io/eslint-plugin-cypress) from 5.0.1 to 5.1.0.
- [Release notes](https://github.com/cypress-io/eslint-plugin-cypress/releases)
- [Commits](https://github.com/cypress-io/eslint-plugin-cypress/compare/v5.0.1...v5.1.0)

---
updated-dependencies:
- dependency-name: eslint-plugin-cypress
  dependency-version: 5.1.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump zod from 3.25.49 to 3.25.51 (#304)

Bumps [zod](https://github.com/colinhacks/zod) from 3.25.49 to 3.25.51.
- [Release notes](https://github.com/colinhacks/zod/releases)
- [Commits](https://github.com/colinhacks/zod/compare/v3.25.49...v3.25.51)

---
updated-dependencies:
- dependency-name: zod
  dependency-version: 3.25.51
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps-dev): bump @stylistic/eslint-plugin from 4.4.0 to 4.4.1 (#307)

Bumps [@stylistic/eslint-plugin](https://github.com/eslint-stylistic/eslint-stylistic/tree/HEAD/packages/eslint-plugin) from 4.4.0 to 4.4.1.
- [Release notes](https://github.com/eslint-stylistic/eslint-stylistic/releases)
- [Changelog](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint-stylistic/eslint-stylistic/commits/v4.4.1/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@stylistic/eslint-plugin"
  dependency-version: 4.4.1
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

bjacobgordon committed:
linter: separates exports from declarations for cleaner diffs (#310)

- `export` next to declarations can add noise to diffs depending on the scenario:
    - `const`: renames
    - `class`: renames, modifying generics, `extends` or `implements`
    - `enum`: renames, modifying cases
    - `type`/`interface`: renames, modifying generics
- keeping exports at the bottom:
    - establishes an easy-to-follow convention
    - balances out "imports at the top"
    - ensures they only show up on diffs when they're actually changed

dependabot[bot] committed:
chore(deps): bump vuetify from 3.8.7 to 3.8.8 (#311)

Bumps [vuetify](https://github.com/vuetifyjs/vuetify/tree/HEAD/packages/vuetify) from 3.8.7 to 3.8.8.
- [Release notes](https://github.com/vuetifyjs/vuetify/releases)
- [Commits](https://github.com/vuetifyjs/vuetify/commits/v3.8.8/packages/vuetify)

---
updated-dependencies:
- dependency-name: vuetify
  dependency-version: 3.8.8
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

bjacobgordon committed:
fix(utilities/Repository): ensures parameters can be added to the url of a drafted issue (#312)

- `searchParams` weren't making it into the clone, so went with getter instead

bjacobgordon committed:
refactor(utilities/Repository): makes it more obvious that error dialog dictates repo issue (#314)

See commits in PR for refactor ops

bjacobgordon committed:
fix(reporting): corrects missing and malformed error details in issue drafts (#313)

* feature(reporting): calls out error details in body of drafted issue

* fix(reporting): ensures all error details are quoted

* fix(reporting): ensures details from error dialog are used in drafted issues

bjacobgordon committed:
feature(reporting): exposes nested error messages to fallback error dialog (#315)

- before, the error dialog only displayed the top-level error message
- this isn't ideal since rethrown errors are usually wrapped by the reason they were rethrown, especially if they were perfectly actionable, but we neglected to embody them with custom UI

stbaione committed:
Merge branch 'integration' into feature-add-input-field-for-guidance-scale

stbaione committed:
Merge branch 'integration' into feature-add-input-field-for-guidance-scale

stbaione committed:
Merge branch 'integration' into feature-add-input-field-for-guidance-scale
